@here/harp-datasource-protocol/lib/Expr.ts

```diff
@@ -119,7 +168,7 @@ export abstract class Expr {
                 if (!(node.length % 2)) {
                     throw new Error("fallback is missing in 'match' expression");
                 }
-                const value = this.fromJSON(node[1]);
+                const value = this.fromJSONInt(node[1], referenceResolverState);
```

Comment:
fromJSONInt sounds a bit strange as name. Please use `fromJSONInternal`, `fromJSONHelper` something else

Reply:
replaced with `parseNode` as to after `parseCall`